docs/operating-scylla/security/encryption-at-rest.rst

```diff
@@ -304,52 +466,107 @@ If you are using AWS KMS to encrypt tables or system information, add the KMS in
+   * ``aws_assume_role_arn`` - If set, any KMS query will first attempt to assume this role.
+   * ``master_key`` - The ID or alias of your AWS KMS key. The key must be generated with an appropriate access policy so that the AWS user has permissions to
+     read the key and encrypt data using that key. This parameter is required.
    * ``certificate`` - The name of the certificate and the path used to identify yourself to the KMS server.
    * ``keyfile`` - The name of the key for the certificate. It is generated together with the certificate.
    * ``truststore`` - The location and key for the truststore to present to the KMS server.
    * ``priority_string`` - The KMS TLS priority string.
    * ``key_cache_expiry`` - Key cache expiry period, after which keys will be re-requested from server. Default is 600s.
-   * ``key_cache_refresh`` - Key cache refresh period - the frequency at which cache is checked for expired entries. Default is 1200s.   
+   * ``key_cache_refresh`` - Key cache refresh period - the frequency at which cache is checked for expired entries. Default is 1200s.
 
-   .. note::
+   Examples:
 
-      Note that either ``endpoint``, ``aws_region`` or ``aws_use_ec2_region`` must be set (one of them is required for connection).
+   .. tabs::
 
-   Example:
+     .. group-tab:: EC2 instance role (recommended)
 
-   .. code-block:: yaml
+        .. code-block:: yaml
 
-       kms_hosts:
-         my-kms1:
-             aws_use_ec2_credentials: true
-             aws_use_ec2_region: true
-             master_key: myorg/MyKey
+            kms_hosts:
+              my-kms1:
+                 aws_use_ec2_credentials: true
+                 aws_use_ec2_region: true
+                 master_key: myorg/MyKey
+
+     .. group-tab:: IAM role (recommended)
+
+        .. code-block:: yaml
+
+            kms_hosts:
+              my-kms1:
+                 aws_use_ec2_credentials: true
+                 aws_use_ec2_region: true
+                 aws_assume_role_arn: arn:aws:iam::123456789012:role/ScyllaDBKMSRole
+                 master_key: myorg/MyKey
+
+     .. group-tab:: IAM user
+
+        .. code-block:: yaml
 
-#. Save the file. 
-#. Drain the node with :doc:`nodetool drain </operating-scylla/nodetool-commands/drain>`
+            kms_hosts:
+              my-kms1:
+                 aws_region: us-east-1
+                 aws_access_key_id: THISISANACCESSKEYID
+                 aws_secret_access_key: THISISASECRETACCESSKEY
+                 master_key: myorg/MyKey
+
+     .. group-tab:: AWS profile
+
+        .. code-block:: yaml
+
+            kms_hosts:
+              my-kms1:
+                 aws_region: us-east-1
+                 aws_profile: user1
+                 master_key: myorg/MyKey
+
+   .. note::
+
+      Note that either ``endpoint``, ``aws_region`` or ``aws_use_ec2_region`` must be set (one of them is required for connection).
```

Comment:
Should we also include that scylla will default to check the standard AWS env vars for ID/key + try load .credentials?